### PR TITLE
grpc

### PR DESCRIPTION
**Pull request by neetinkandhare**
## Motivation and Context

Fixes #

## Checklist:

- [] I have added a [Signed-off-by](https://github.com/IBMResilient/resilient-community-apps/blob/master/CONTRIBUTING.md)
- [] Either no new documentation is required by this change, OR I added new documentation
- [] Either no new tests are required by this change, OR I added new tests
- [] I have run pep8 and pylint. I have cleaned up all valid errors and warnings in code I have added or modified. These tools may generate false positives. Don't be worried about ignoring some errors or warnings. The goal is clean, consistent, and readable code.

Signed-off-by:

### CODE REVIEW

File: fn_grpc_interface/fn_grpc_interface/components/function_grpc.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+                            stub_method = self._get_method_from_stub_object(stub, _grpc_communication_type, stub_name=_grpc_rpc_stub_method_name)
+                            if grpc_function_json_data:
+                                response_received = stub_method[1](grpc_request_tuple[1](**grpc_function_json_data))
+                    except grpc.RpcError as rpc_err:
+                        status_code = rpc_err.code()
+                        log.info("{}\n connection status :{} \n name : {}\n value : {}".format(rpc_err,rpc_err.details(),status_code.name,status_code.value))
+                    else:
+                        log.info("Response received from the server : {}".format(str(response_received)))
+                else:
+                    logging.info("Connection type is Server_Stream or client_stream,Bidirectional or other\n \
```
Comment (mscherfling) [inline fn_grpc_interface/fn_grpc_interface/components/function_grpc.py]: raise an error on this condition if we're not implementing it.

File: fn_grpc_interface/fn_grpc_interface/util/config.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+
+def config_section_data():
+    """Produce the default configuration section for app.config,
+       when called by `resilient-circuits config [-c|-u]`
+    """
+    config_data = u"""[fn_grpc]
+     #package_Name = communication type(i.e unary,server_stream,client_stream,bidirection_stream),secure connection type(i.e None,TLS,SSL,OAuth2),certificate_path or google API Tocken,Interface pb2 files parent directory
+helloword=unary,None,None
+interface_dir = /home/neetin/gRPC_Project/
```
Comment (mscherfling) [inline fn_grpc_interface/fn_grpc_interface/util/config.py]: take your name out of this so it's more generic

File: fn_grpc_interface/fn_grpc_interface/util/config.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+calculator=unary,None,None
```
Comment (mscherfling) [inline fn_grpc_interface/fn_grpc_interface/util/config.py]: for all examples, specify them as comments so not to confuse with actual settings:
#helloword=unary,None,None
#calculator=unary,None,None

File: fn_grpc_interface/fn_grpc_interface/util/selftest.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
Comment (mscherfling) [inline fn_grpc_interface/fn_grpc_interface/util/selftest.py]: aren't you going to delete this?

File: fn_grpc_interface/setup.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from setuptools import setup, find_packages
+
+setup(
+    name='fn_grpc_interface',
+    version='1.0.0',
+    license='<<insert here>>',
```
Comment (mscherfling) [inline fn_grpc_interface/setup.py]: complete all these fields

File: fn_grpc_interface/setup.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    author='<<your name here>>',
+    author_email='you@example.com',
+    url='<<your company url>>',
+    description="Resilient Circuits Components for 'fn_grpc_interface'",
+    long_description="Resilient Circuits Components for 'fn_grpc_interface'",
+    install_requires=[
+        'resilient_circuits>=30.0.0'
```
Comment (mscherfling) [inline fn_grpc_interface/setup.py]: add the grpc libraries

File: fn_grpc_interface/tests/test_function_grpc.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
```
Comment (mscherfling) [inline fn_grpc_interface/tests/test_function_grpc.py]: if this isn't used, delete it

File: fn_grpc_interface/fn_grpc_interface/LICENSE [hunk 8]
```diff
@@ -0,0 +1 @@
+<<PUT YOUR LICENSE TEXT HERE>>
```
[the remark this answers is not in the document]
Comment (neetinkandhare) [reply fn_grpc_interface/fn_grpc_interface/LICENSE]: ok

File: fn_grpc_interface/tests/test_function_grpc.py (hunk 7 above)
Comment (neetinkandhare) [reply fn_grpc_interface/tests/test_function_grpc.py]: ok

File: fn_grpc_interface/setup.py (hunk 6 above)
Comment (neetinkandhare) [reply fn_grpc_interface/setup.py]: ok

File: fn_grpc_interface/setup.py (hunk 5 above)
Comment (neetinkandhare) [reply fn_grpc_interface/setup.py]: ok

File: fn_grpc_interface/fn_grpc_interface/util/selftest.py (hunk 4 above)
Comment (neetinkandhare) [reply fn_grpc_interface/fn_grpc_interface/util/selftest.py]: deleted.

File: fn_grpc_interface/fn_grpc_interface/util/config.py (hunk 3 above)
Comment (neetinkandhare) [reply fn_grpc_interface/fn_grpc_interface/util/config.py]: ok

File: fn_grpc_interface/fn_grpc_interface/components/function_grpc.py [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+        self.options = opts.get("fn_grpc", {})
+
+
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_grpc", {})
+
+
+    def _get_interface_file_names(self, base_dir=None, files_dir=None):
```
[the remark this answers is not in the document]
Comment (neetinkandhare) [reply fn_grpc_interface/fn_grpc_interface/components/function_grpc.py]: ok

File: fn_grpc_interface/fn_grpc_interface/components/function_grpc.py [hunk 10]
```diff
@@ -0,0 +1,195 @@
+# -*- coding: utf-8 -*-
```
[the remark this answers is not in the document]
Comment (neetinkandhare) [reply fn_grpc_interface/fn_grpc_interface/components/function_grpc.py]: ok

File: fn_grpc_interface/fn_grpc_interface/components/function_grpc.py [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+from resilient_circuits import ResilientComponent, function, handler, StatusMessage, FunctionResult, FunctionError
+import sys
+import os
+import json
+import grpc
+import grpc_tools
+import re
+import inspect
+class FunctionComponent(ResilientComponent):
+    """Component that implements Resilient function 'function_grpc"""
```
[the remark this answers is not in the document]
Comment (neetinkandhare) [reply fn_grpc_interface/fn_grpc_interface/components/function_grpc.py]: ok

File: fn_grpc_interface/README.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
+Replace this with your own readme.
```
[the remark this answers is not in the document]
Comment (neetinkandhare) [reply fn_grpc_interface/README.md]: ok

**Pull request closed without merging**